pkg/kf/testutil/assertions.go

```diff
@@ -121,6 +121,54 @@ func AssertContainsAll(t Failable, haystack string, needles []string) {
 	}
 }
 
+// AssertConsistOf fails if the container (slice or array, string, map, or
+// channel) does not consist of the given needles. Order does not matter, but
+// length does. If haystack is a map, the keys are evaluated. If haystack is a
+// channel, then values are read from the channel (therefore it destroys the
+// input).
+func AssertConsistOf(t Failable, fieldName string, haystack interface{}, needles ...interface{}) {
+	var idxF func(int) interface{}
+
+	v := reflect.ValueOf(haystack)
+	switch reflect.TypeOf(haystack).Kind() {
+	case reflect.Slice, reflect.Array:
+		idxF = func(i int) interface{} {
+			return v.Index(i).Interface()
+		}
+	case reflect.String:
+		str := haystack.(string)
+		idxF = func(i int) interface{} {
+			return interface{}(str[i])
+		}
+	case reflect.Map:
+		keys := v.MapKeys()
+		idxF = func(i int) interface{} {
+			return v.MapIndex(keys[i]).Interface()
+		}
+	case reflect.Chan:
```

Comment:
I'm hesitant to build out additional testing functionality we don't need yet due to #56